src/domain/consents/generateChallenge.ts

```diff
@@ -54,7 +54,7 @@ export function isConsentRequestInitiatedByValidSource (
   request: Request,
   consent: Consent): boolean {
   const fspiopSource = request.headers[Enum.Http.Headers.FSPIOP.SOURCE]
-  return (consent && consent.initiatorId === fspiopSource)
+  return (consent?.initiatorId === fspiopSource)
```

Comment:
If consent is null AND fsoiopSource is null, this should evaluate to `true` which is probably not what you want.... You should probably add a test to verify this before changing it back.